I `pip install`'d the `nextmv` folder of the `merschformann/support-amd64-python-app-target` branch into a virtual environment and then ran this from [the root of the `python-cuopt-routing` community app](https://github.com/nextmv-io/community-apps/tree/rjo/python-cuopt-routing/python-cuopt-routing).


```python
import os
from nextmv import cloud


client = cloud.Client(api_key=os.getenv("XYZZY-XYZZY-XYZZY-XYZZY"))
app = cloud.Application(client=client, id="xyzzy")

app.push(verbose=True)
```

This is what it gave me:

```txt
.venv) ryan@part:~/.../community-apps/python-cuopt-routing$ python push.py 
/home/ryan/src/github.com/nextmv-io/community-apps/python-cuopt-routing/.venv/lib/python3.13/site-packages/nextmv/cloud/acceptance_test.py:261: DeprecationWarning: ToleranceType: ToleranceType is deprecated and will be removed in a future version. Please use MetricToleranceType instead. This functionality will be removed in a future release
  deprecated(
/home/ryan/src/github.com/nextmv-io/community-apps/python-cuopt-routing/.venv/lib/python3.13/site-packages/nextmv/cloud/acceptance_test.py:261: DeprecationWarning: ToleranceType: ToleranceType is deprecated and will be removed in a future version. Please use MetricToleranceType instead. This functionality will be removed in a future release
  deprecated(
/home/ryan/src/github.com/nextmv-io/community-apps/python-cuopt-routing/.venv/lib/python3.13/site-packages/nextmv/cloud/acceptance_test.py:261: DeprecationWarning: ToleranceType: ToleranceType is deprecated and will be removed in a future version. Please use MetricToleranceType instead. This functionality will be removed in a future release
  deprecated(
💽 Starting build for Nextmv application.
Traceback (most recent call last):
  File "/home/ryan/src/github.com/nextmv-io/community-apps/python-cuopt-routing/push.py", line 8, in <module>
    app.push(verbose=True)
    ~~~~~~~~^^^^^^^^^^^^^^
  File "/home/ryan/src/github.com/nextmv-io/community-apps/python-cuopt-routing/.venv/lib/python3.13/site-packages/nextmv/cloud/application.py", line 2523, in push
    manifest = Manifest.from_yaml(app_dir)
  File "/home/ryan/src/github.com/nextmv-io/community-apps/python-cuopt-routing/.venv/lib/python3.13/site-packages/nextmv/manifest.py", line 1091, in from_yaml
    return cls.from_dict(raw_manifest)
           ~~~~~~~~~~~~~^^^^^^^^^^^^^^
  File "/home/ryan/src/github.com/nextmv-io/community-apps/python-cuopt-routing/.venv/lib/python3.13/site-packages/nextmv/base_model.py", line 54, in from_dict
    return cls(**data)
  File "/home/ryan/src/github.com/nextmv-io/community-apps/python-cuopt-routing/.venv/lib/python3.13/site-packages/pydantic/main.py", line 250, in __init__
    validated_self = self.__pydantic_validator__.validate_python(data, self_instance=self)
pydantic_core._pydantic_core.ValidationError: 2 validation errors for Manifest
runtime
  Input should be 'ghcr.io/nextmv-io/runtime/default:latest', 'ghcr.io/nextmv-io/runtime/python:3.11', 'ghcr.io/nextmv-io/runtime/java:latest', 'ghcr.io/nextmv-io/runtime/pyomo:latest' or 'ghcr.io/nextmv-io/runtime/hexaly:latest' [type=enum, input_value='ghcr.io/nextmv-io/runtime/cuopt:latest', input_type=str]
    For further information visit https://errors.pydantic.dev/2.12/v/enum
python.version
  Input should be a valid string [type=string_type, input_value=3.12, input_type=float]
    For further information visit https://errors.pydantic.dev/2.12/v/string_type
```

--- a/nextmv/nextmv/__init__.py
+++ b/nextmv/nextmv/__init__.py
@@ -21,6 +21,7 @@ from .manifest import Manifest as Manifest
 from .manifest import ManifestBuild as ManifestBuild
 from .manifest import ManifestOption as ManifestOption
 from .manifest import ManifestPython as ManifestPython
+from .manifest import ManifestPythonArch as ManifestPythonArch
 from .manifest import ManifestPythonModel as ManifestPythonModel
 from .manifest import ManifestRuntime as ManifestRuntime
 from .manifest import ManifestType as ManifestType

--- a/nextmv/nextmv/manifest.py
+++ b/nextmv/nextmv/manifest.py
@@ -172,6 +172,39 @@ class ManifestRuntime(str, Enum):
     """
 
 
+class ManifestPythonArch(str, Enum):
+    """
+    Target architecture for bundling Python apps.
+
+    You can import the `ManifestPythonArch` class directly from `nextmv`:
+
+    ```python
+    from nextmv import ManifestPythonArch
+    ```
+
+    Attributes
+    ----------
+    ARM64 : str
+        ARM 64-bit architecture.
+    AMD64 : str
+        AMD 64-bit architecture.
+
+    Examples
+    --------
+    >>> from nextmv import ManifestPythonArch
+    >>> arch = ManifestPythonArch.ARM64
+    >>> arch
+    <ManifestPythonArch.ARM64: 'arm64'>
+    >>> str(arch)
+    'arm64'
+    """
+
+    ARM64 = "arm64"
+    """ARM 64-bit architecture."""
+    AMD64 = "amd64"
+    """AMD 64-bit architecture."""
+
+
 class ManifestBuild(BaseModel):
     """
     Build-specific attributes.
@@ -329,7 +362,7 @@ class ManifestPython(BaseModel):
     Contains (additional) Python dependencies that will be bundled with the
     app.
     """
-    arch: Optional[str] = None
+    arch: Optional[ManifestPythonArch] = None
     """The architecture this model is meant to run on. One of "arm64" or "amd64". Uses
     "arm64" if not specified."""
     version: Optional[Union[str, float]] = None

--- a/nextmv/nextmv/manifest.py
+++ b/nextmv/nextmv/manifest.py
@@ -166,6 +166,10 @@ class ManifestRuntime(str, Enum):
     Based on the python runtime, it provisions (pre-installs) the Hexaly solver
     to run Python applications.
     """
+    CUOPT = "ghcr.io/nextmv-io/runtime/cuopt:latest"
+    """
+    A runtime providing the NVIDIA cuOpt solver.
+    """
 
 
 class ManifestBuild(BaseModel):

--- a/nextmv/nextmv/manifest.py
+++ b/nextmv/nextmv/manifest.py
@@ -49,7 +49,7 @@ from enum import Enum
 from typing import Any, Optional, Union
 
 import yaml
-from pydantic import AliasChoices, Field
+from pydantic import AliasChoices, Field, field_validator
 
 from nextmv.base_model import BaseModel
 from nextmv.input import InputFormat
@@ -332,7 +332,7 @@ class ManifestPython(BaseModel):
     arch: Optional[str] = None
     """The architecture this model is meant to run on. One of "arm64" or "amd64". Uses
     "arm64" if not specified."""
-    version: Optional[str] = None
+    version: Optional[Union[str, float]] = None
     """The Python version this model is meant to run with. Uses "3.11" if not specified.
     """
     model: Optional[ManifestPythonModel] = None
@@ -341,6 +341,17 @@ class ManifestPython(BaseModel):
     As handled via mlflow. This information is used to load the decision model
     from the app bundle.
     """
+
+    @field_validator("version", mode="before")
+    @classmethod
+    def validate_version(cls, v: Optional[Union[str, float]]) -> Optional[str]:
+        # We allow the version to be a float in the manifest for convenience, but we want
+        # to store it as a string internally.
+        if v is None:
+            return None
+        if isinstance(v, float):
+            return str(v)
+        return v
 
 
 class ManifestOptionUI(BaseModel):
@@ -1035,7 +1046,12 @@ class Manifest(BaseModel):
 
     def model_post_init(self, __context) -> None:
         if self.entrypoint is None:
-            if self.runtime in (ManifestRuntime.PYTHON, ManifestRuntime.HEXALY, ManifestRuntime.PYOMO):
+            if self.runtime in (
+                ManifestRuntime.PYTHON,
+                ManifestRuntime.HEXALY,
+                ManifestRuntime.PYOMO,
+                ManifestRuntime.CUOPT,
+            ):
                 self.entrypoint = "./main.py"
             elif self.runtime == ManifestRuntime.DEFAULT:
                 self.entrypoint = "./main"

